src/lib/api-response.ts

```
@@ -0,0 +1,273 @@
+export function validateQueryParams(
+  url: URL,
+  optional: Record<string, 'string' | 'number' | 'boolean'> = {}
+): Record<string, unknown> {
+  const result: Record<string, unknown> = {};
+  
+  for (const [key, type] of Object.entries(optional)) {
+    const value = url.searchParams.get(key);
+    if (value !== null) {
+      switch (type) {
+        case 'number':
+          const num = parseInt(value, 10);
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)